sdk/java/src/ai/picovoice/picovoice/Picovoice.java

```diff
@@ -129,20 +168,36 @@ public void delete() {
      * @throws PicovoiceException if there is an error while processing the audio frame.
      */
     public void process(short[] pcm) throws PicovoiceException {
+        if (porcupine == null || rhino == null) {
+            throw new PicovoiceInvalidStateException("Cannot process frame - resources have been released");
+        }
+
+        if (pcm == null) {
+            throw new PicovoiceInvalidArgumentException("Passed null frame to Picovoice process.");
+        }
+
+        if (pcm.length != getFrameLength()) {
+            throw new PicovoiceInvalidArgumentException(
+                    String.format("Picovoice process requires frames of length %d. " +
+                            "Received frame of size %d.", getFrameLength(), pcm.length));
+        }
+
         try {
             if (!isWakeWordDetected) {
                 isWakeWordDetected = (porcupine.process(pcm) == 0);
-                if (isWakeWordDetected) {
+                if (isWakeWordDetected && wakeWordCallback != null) {
```

Comment:
can this be `null`?

Reply:
i guess in the past yes, but now that we are checking both `wakeWordCallback` and `inferenceCallback` in the constructor, not anymore. I removed checking for `null` callbacks in the process functions.